src/OpenTK/Platform/X11/XI2MouseKeyboard.cs

```diff
@@ -167,11 +167,32 @@ internal static bool IsSupported(IntPtr display)
             return false;
         }
 
+        private bool GetIndicatorState(string name)
+        {
+            bool state = false;
+            bool lockState = false;
+            int ndx = 0;
+            IntPtr map = new IntPtr ();
+            bool real = false;
+        
+            IntPtr atom = Functions.XInternAtom(API.DefaultDisplay, name, true);
```

Comment:
Can you cache these atoms on a class field when creating the class, XInternAtom isn't cheep to call.